msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,131 @@
+
+To ensure intentional and deliberate customer experiences with your app, you can specify your app's runtime requirements in the app manifest to tailor its behavior in applicable Microsoft 365 hosts, or omit it from surfacing in contexts that you're not ready to support. Describing your app's runtime requirements in the app manifest also helps ensure your production-ready app experience will reach wider audiences in additional Microsoft 365 host applications in the future.
+
+Specifying your app's runtime requirements is useful in scenarios such as:
+
+- **One-way dependencies**: If your app includes both a tab and message extension, and the tab functions as a settings page for users to configure message extension functionality, you can ensure that your settings tab doesn't load in hosts that don't support your message extension.
+
+- **Mutual dependencies:** When your app has specific app components that must be loaded together in order to function properly, you can ensure your app is *only available* in Microsoft 365 hosts that support all components.  
+
+- **Capability requirements:** When your app has component parts with runtime requirements that aren't supported across all applicable Microsoft 365 hosts, you can ensure those components are *seamlessly omitted* (rather than loaded, yet not functional) from your app experience running in those hosts.
+
+## Microsoft 365 host support
+
+Only a subset of Microsoft 365 host applications supports the ability to specify runtime requirements in the app manifest. This support will expand over time. The following hosts ensure only applicable apps and their applicable components are made available to users:
+
+|Microsoft 365 host application| Web | Desktop | Mobile |
+|---|---|---|---|
+| Microsoft Teams| ✔️| ✔️| ✔️|
+| Microsoft Outlook| ✔️| ✔️| ✔️|
+
+## Specify relationships between components of your app (`elementRelationshipSet`)
+
+You can specify relationships among the individual components of your app by including an [`elementRelationshipSet`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipset) in your app manifest. Use this object to specify both [one-way dependencies](#one-way-dependencies) and [mutual dependencies](#mutual-dependencies) among app components.
+
+> [!NOTE]
+> A given pair of app components can only be grouped once, either by `oneWayDependency` or `mutualDependency`. Similarly, cyclical `oneWayDependencies`—where, for example, components *A* and *B* mutually declare each other as a one-way dependency—aren't allowed and will fail manifest validation.
```

Comment:
Are we referring to bots / tabs as components or elements? We can be consistent in calling them elements.

Reply:
@juna-gupta I'm using the descriptive phrase "component of an app" to bridge the conceptual gap between Teams / MetaOS terminology. Teams docs generalize tabs/bots/MEs etc as "app capabilities"; MetaOS WG has been unofficially calling them "elements". I'm trying to use the generic phrase "app component" to describe RT for both Teams and new-to-M365 dev audiences

Reply:
@erikadoyle, I suggest we use the term app capabilities.